docs/spec/v2.rst

```diff
@@ -312,6 +338,61 @@ For example, the JSON object below encodes three attributes named
         "baz": [1, 2, 3, 4]
     }
 
+.. _spec_v2_dimensions:
+
+Dimensions
+----------
+
+Groups and arrays can be associated with optional dimension names. This feature
+is intended to facilitate self-described datasets.
+
+Dimensions are required to be consistent. Any dimensions set on an array
+(any non-``null`` value), MUST also be defined on an ancestor group. Dimension
+sizes can be overwritten in descendent groups, but the size of each named
+dimensions on an array MUST match the size of that dimension on the most direct
+ancestor group on which it is defined.
```

Comment:
I think I'm going to change this, to make group dimensions and consistency entirely optional:
```
If dimensions are set in a group, their sizes on all contained arrays
are REQUIRED to be consistent. Dimension sizes can be overwritten
in descendant groups, but the size of each named dimension (any
non-`null` value) on an array MUST match the size of that dimension
on the most direct ancestor group on which it is defined.
```